core/solver/cg.hpp

```diff
@@ -41,6 +41,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "core/base/types.hpp"
 #include "core/log/logger.hpp"
 #include "core/matrix/identity.hpp"
+#include "core/stop/combined.hpp"
 #include "core/stop/criterion.hpp"
```

Comment:
Technically you are missing `#include <vector>` here and in the following classes. It's better to not depend on third party headers being provided by included headers (except when it makes sense and it is clear that this would always be the case).